processing/src/main/java/org/apache/druid/query/aggregation/Aggregator.java

```diff
@@ -42,6 +42,12 @@
 {
   void aggregate();
 
+  default long aggregateWithSize()
```

Comment:
Thanks for the reminder, @abhishekagarwal87. I am adding javadocs wherever missing.

Reply:
what is the contract of this value, I don't see the word estimate in here, but think it probably should be... should implementors over-estimate if exact sizing is not possible or is under-estimating fine? Should there be a warning that the default estimate is used? (i imagine this would be very noisy if it is done per aggregate call... so don't really recommend doing it here or anything...)

Reply:
> Should there be a warning that the default estimate is used?

It would make sense to give this warning when `factorizeWithSize` is overridden but `aggregateWithSize` is not. In such a case, we might be significantly underestimating the memory usage.

As you said, doing it here might be noisy. A viable approach could be to have `factorizeWithSize` return a wrapper Aggregator which does not allow the regular `aggregate` (will be addressed in the subsequent PR).